File: 217-contains-duplicate/contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    // Given an integer array nums, return true if any value appears at least twice 
    // false if every element is distinct.
    let hm = new Map()
    for (let num of nums) {
        hm.set(num, (hm.get(num) || 0) + 1)
    }
    for (let val of hm.values()) {
        if (val > 1) {
            return true
        }
    }
    return false

};

File: 217-contains-duplicate/contains-duplicate.rb
# @param {Integer[]} nums
# @return {Boolean}
def contains_duplicate(nums)
    # Input: array of numbers nums[]
    # Output: boolean

    nums_frequency = Hash.new(0)
    nums.each do |num|
        if nums_frequency.has_key?(num)
            return true
        else
            nums_frequency[num] += 1
        end
    end

    false
end